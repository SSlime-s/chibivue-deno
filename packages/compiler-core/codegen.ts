import type { ParsedContent } from "./parse.ts";

export function generate({ tag, props, textContent }: ParsedContent): string {
  return `
return () => {
  const { h } = ChibiVue;
  return h(
    "${tag}", { ${
    Object.entries(props).map(([key, value]) => `${key}: "${value}"`).join(
      ",",
    )
  } }, ["${textContent}"]);
}`;
}
