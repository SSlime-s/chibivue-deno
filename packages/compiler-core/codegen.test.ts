import { assertSnapshot } from "@std/testing/snapshot";
import { generate } from "./codegen.ts";

Deno.test("generate できる", async (t) => {
  const parsed = {
    tag: "div",
    props: { id: "app" },
    textContent: "Hello, ChibiVue!",
  };
  await assertSnapshot(t, generate(parsed));
});
