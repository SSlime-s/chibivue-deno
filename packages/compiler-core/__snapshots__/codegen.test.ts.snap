export const snapshot = {};

snapshot[`generate できる 1`] = `
'
return () => {
  const { h } = ChibiVue;
  return h(
    "div", { id: "app" }, ["Hello, ChibiVue!"]);
}'
`;
